### PR TITLE
bpo-26053: Correct assertion in pdb test

### PR DESCRIPTION
**Pull request by iritkatriel**
https://bugs.python.org/issue26053

### CODE REVIEW

**Comment by iritkatriel** [comment]
This passed on a mac (where master currently fails). How do we check why the CI didn't find the issue?

**Comment by iritkatriel** [comment]
The previous PR had been open for a long time. Could it be that the tests used to pass, and now when it was merged they were not executed again?

File: Lib/test/test_pdb.py [hunk 1]
```diff
@@ -1453,9 +1453,9 @@ def test_issue26053(self):
             quit
         """
         stdout, stderr = self.run_pdb_script(script, commands)
-        output = '\n'.join([x.strip() for x in stdout.splitlines()])
-        self.assertIn("Restarting main.py with arguments:\na b c", output)
-        self.assertIn("Restarting main.py with arguments:\nd e f", output)
+        res = '\n'.join([x.strip() for x in stdout.splitlines()])
+        self.assertRegex(res, "Restarting .* with arguments:\na b c")
+        self.assertRegex(res, "Restarting .* with arguments:\nd e f")
```
Comment (vstinner) [inline Lib/test/test_pdb.py:1458]: Can you use script? Maybe you need os.path.basename(), I don't know pdb.

Comment (iritkatriel) [reply Lib/test/test_pdb.py:1458]: what is script?  The important part of the test is the "a b c" and "d e f"

**Comment by iritkatriel** [comment]
> The previous PR had been open for a long time. Could it be that the tests used to pass, and now when it was merged they were not executed again?

I think this is the reason the CI missed it - I can see that new open PRs are blocked now with this error.

File: Lib/test/test_pdb.py (hunk 1 above)
Comment (iritkatriel) [reply Lib/test/test_pdb.py:1458]: The filename comes from Bdb.canonic:

    ```
def canonic(self, filename):
        """Return canonical form of filename.

        For real filenames, the canonical form is a case-normalized (on
        case insensitive filesystems) absolute path.  'Filenames' with
        angle brackets, such as "<stdin>", generated in interactive
        mode, are returned unchanged.
        """
        if filename == "<" + filename[1:-1] + ">":
            return filename
        canonic = self.fncache.get(filename)
        if not canonic:
            canonic = os.path.abspath(filename)
            canonic = os.path.normcase(canonic)
            self.fncache[filename] = canonic
        return canonic
```

Comment (iritkatriel) [reply Lib/test/test_pdb.py:1458]: I changed it to use canonic(). It passes on the mac.

Comment (iritkatriel) [reply Lib/test/test_pdb.py:1458]: Oh, I see what you mean - no, script is the code (contents of the script). The name is main.py, hard-coded in run_pdb_script()

**Review by vstinner — APPROVED** [review]
LGTM if the tests pass ;-)

**Comment by iritkatriel** [comment]
> LGTM if the tests pass ;-)

I think they fail on windows.

**Comment by iritkatriel** [comment]
> > LGTM if the tests pass ;-)
> 
> I think they fail on windows.

c:\ vs C:\

**Comment by iritkatriel** [comment]
I reverted it back - it's not simple to get to the pdb filename from here (the pdb instance is destroyed in run_pdb_script and we only have the stdin/stdout in the test function). And, as I said, the filename is not the point of this test anyway, the point is to see that the correct args are picked up.

**Pull request merged**